src/error.rs

```diff
@@ -101,7 +101,7 @@ impl Error for FormatterError {
     fn source(&self) -> Option<&(dyn Error + 'static)> {
         match self {
             Self::Idempotence => None,
-            Self::Internal(_, source) => source.as_ref().map(|e| e as &dyn Error),
+            Self::Internal(_, source) => source.as_ref().map(|e| &**e),
```

Comment:
Isn't `&**e` equivalent to `*e`? Is this related to [this comment](https://tweag.slack.com/archives/CKMLS3094/p1678342865049809)?

Reply:
`e` is an `&Box<dyn Error>`. `*e` results in `expected reference, found struct Box`. We need to deref that `Box` with an extra `*`. Now we have a `dyn Error`. But since the function wants to return an `&dyn Error` (wrapped in an `Option`), we add an `&`.